lib/enmeshed/relationship_template.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+module Enmeshed
+  class RelationshipTemplate
+    # The app does not allow users to scan expired templates.
+    # However, previously scanned and then expired templates can still be submitted,
+    # resulting in the app silently doing nothing. CodeHarbor would still accept Relationships for expired templates if sent by the app.
+    # To minimize the risk of a template expiring before submission, we set the validity to 12 hours.
+    VALIDITY_PERIOD = 12.hours
+    # The display name of the service as shown in the enmeshed app.
+    DISPLAY_NAME = Settings.omniauth&.nbp&.enmeshed&.display_name
+    # These attributes are mandatory in the app and must be provided.
+    # See https://enmeshed.eu/integrate/attribute-values for more attributes.
+    REQUIRED_ATTRIBUTES = %w[GivenName Surname EMailAddress AffiliationRole].freeze
+
+    attr_reader :expires_at, :nbp_uid
+
+    def initialize(options = {})
+      skip_fetch = options.delete(:skip_fetch)
+      case options.keys
+        when [:content]
+          template = options[:content]
+          @truncated_reference = template[:truncatedReference]
+          populate_from_existing template
+        when [:truncated_reference]
+          @truncated_reference = options[:truncated_reference]
+          fetch_existing unless skip_fetch
```

Comment:
Do we need fetch_existing? I could not find an initialization where truncated_reference is used with skip_fetch=false.

Reply:
We probably don't need it right now, I agree. At the moment, there is no usage with `RelationshipTemplate.new(truncated_reference: 'tr', skip_fetch: false)`. 

My first idea was to add it lazy, so that accessing the `nbp_uid` or `expires_at` would trigger fetching the record if not present. Then, I decided to move that to the initialize, to make it more similar with the other initializers.

We could, of course, remove the code related to fetching (or enable it 😁). This would remove some code paths not used during regular operation, but would (on the other hand) also require to invest some time in the future when requiring it. Both options are not ideal, and I would normally tend to drop it. However, given the complexity of the domain and API, it might rewarding to have working code 😉

Reply:
Okay. I think the PR is ready to be merged then.